include/nbl/video/IGPUObjectFromAssetConverter.h

```diff
@@ -591,76 +710,189 @@ auto IGPUObjectFromAssetConverter::create(const asset::ICPUMesh** const _begin,
 
     return res;
 }
+#endif
 
 auto IGPUObjectFromAssetConverter::create(const asset::ICPUImage** const _begin, const asset::ICPUImage** const _end, const SParams& _params) -> created_gpu_object_array<asset::ICPUImage>
 {
     const auto assetCount = std::distance(_begin, _end);
     auto res = core::make_refctd_dynamic_array<created_gpu_object_array<asset::ICPUImage> >(assetCount);
 
+    bool needToGenMips = false;
+    core::vector<IGPUCommandBuffer::SImageMemoryBarrier> imgMemBarriers;
+    imgMemBarriers.reserve(assetCount);
+
+    core::unordered_map<const asset::ICPUImage*, core::smart_refctd_ptr<IGPUBuffer>> img2gpubuf;
+    for (ptrdiff_t i = 0u; i < assetCount; ++i)
+    {
+        const asset::ICPUImage* cpuimg = _begin[i];
+        if (cpuimg->getRegions().size() == 0ull)
+            continue;
+
+        auto gpubuf = _params.device->createDeviceLocalGPUBufferOnDedMem(cpuimg->getBuffer()->getSize());
+        img2gpubuf.insert({ cpuimg, std::move(gpubuf) });
+
+        if (asset::isIntegerFormat(cpuimg->getCreationParameters().format))
+            needToGenMips = true;
+    }
+    core::smart_refctd_ptr<IGPUCommandPool> pool;
+    core::smart_refctd_ptr<IGPUFence> fence;
+    core::smart_refctd_ptr<IGPUCommandBuffer> cmdbuf;
+
+    if (img2gpubuf.size())
+    {
+        pool = _params.device->createCommandPool(_params.transferQueue->getFamilyIndex(), IGPUCommandPool::ECF_RESET_COMMAND_BUFFER_BIT);
+        fence = _params.device->createFence(static_cast<IGPUFence::E_CREATE_FLAGS>(0));
+        _params.device->createCommandBuffers(pool.get(), IGPUCommandBuffer::EL_PRIMARY, 1u, &cmdbuf);
+
+        cmdbuf->begin(IGPUCommandBuffer::EU_ONE_TIME_SUBMIT_BIT);
+
+        for (auto& it : img2gpubuf)
+        {
+            auto* cpuimg = it.first;
+            auto* cpubuf = cpuimg->getBuffer();
+            assert(cpubuf);
+            auto& gpubuf = it.second;
+
+            asset::SBufferRange<IGPUBuffer> bufrng;
+            bufrng.buffer = gpubuf;
+            bufrng.offset = 0u;
+            bufrng.size = gpubuf->getSize();
+
+            _params.device->updateBufferRangeViaStagingBuffer(cmdbuf.get(), fence.get(), _params.transferQueue, bufrng, cpubuf->getPointer());
+        }
```

Comment:
if  you adhere to 
https://github.com/Devsh-Graphics-Programming/Nabla/pull/28/files/236b8ecdf029cd54251c4f120fd98300c23b6fb7..91b055e35d412c474cc3c8bec8802ffd2681bd9d#r614277697

code will be more dry.

also you dont need to signal a fence here at all, only do a "split barrier" (using an Event) if the transfer and compute queue are the same queue